sdk/js/src/utils/consts.ts

```diff
@@ -790,51 +811,33 @@ export function coalesceChainName(chain: ChainId | ChainName): ChainName {
 export function isEVMChain(
   chain: ChainId | ChainName
 ): chain is EVMChainId | EVMChainName {
-  let chainId = coalesceChainId(chain);
-  if (
-    chainId === CHAIN_ID_ETH ||
-    chainId === CHAIN_ID_BSC ||
-    chainId === CHAIN_ID_AVAX ||
-    chainId === CHAIN_ID_POLYGON ||
-    chainId === CHAIN_ID_OASIS ||
-    chainId === CHAIN_ID_AURORA ||
-    chainId === CHAIN_ID_FANTOM ||
-    chainId === CHAIN_ID_KARURA ||
-    chainId === CHAIN_ID_ACALA ||
-    chainId === CHAIN_ID_KLAYTN ||
-    chainId === CHAIN_ID_CELO ||
-    chainId === CHAIN_ID_MOONBEAM ||
-    chainId === CHAIN_ID_NEON ||
-    chainId === CHAIN_ID_ARBITRUM ||
-    chainId === CHAIN_ID_OPTIMISM ||
-    chainId === CHAIN_ID_GNOSIS ||
-    chainId === CHAIN_ID_BASE ||
-    chainId === CHAIN_ID_SEPOLIA
-  ) {
-    return isEVM(chainId);
+  const chainName = coalesceChainName(chain);
+  if (EVMChainNames.includes(chainName)){
+    return isEVM(chainName as EVMChainName);
   } else {
-    return notEVM(chainId);
+    return notEVM(chainName as NonEVMChainName);
```

Comment:
the `notEVM` was used to ensure that the membership check above was performed correctly (i.e. if you missed an EVM chain ID in the conditional, it would complain), but with the array version, the check should just be correct by construction. Thus we should be able to just remove the if-then-else block entirely and return `EVMChainNames.includes(chainName)`.
That way, we can get rid of `notEVM` and `NotEVMChainName` entirely

Reply:
took the if/else out and type definition I added. Should we also remove the `isEVM`/`notEVM` and `evm_chain_subset`? they're not exported and don't appear used